exporter/exporterhelper/queued_retry.go

```diff
@@ -273,9 +273,9 @@ func NewDefaultRetrySettings() RetrySettings {
 	}
 }
 
-func createSampledLogger(logger *zap.Logger) *zap.Logger {
-	if logger.Core().Enabled(zapcore.DebugLevel) {
-		// Debugging is enabled. Don't do any sampling.
+func createSampledLogger(logger *zap.Logger, lCfg SampledLoggerSettings) *zap.Logger {
```

Comment:
@codeboten, do you know why we create a sampled logger here? Shouldn't loggers be created based on the telemetry configuration alone? Instead of this PR, I would consider removing this sampled logger altogether.

Reply:
This was added in #2020 by @tigrannajaryan. The PR description says

> All logging is heavily sampled to avoid flooding the lo

Reply:
Right, and shouldn't this be a collector-wide setting, instead of something exclusively to this helper?

Reply:
I think this was meant to be specific to this helper to avoid flooding the logs in the case where all requests fail, and you log once per request. The [contributing guidelines][1] say

> Use logging carefully for events that can happen frequently to avoid flooding the logs. Avoid outputting logs per a received or processed data item since this can amount to very large number of log entries (Collector is designed to process many thousands of spans and metrics per second). For such high-frequency events instead of logging consider adding an internal metric and increment it when the event happens.

I guess at the time this was written metrics were not really available and thus we went with sampled logs.

(Not sure if this is a good reason today to keep this, just trying to reconstruct the history of this to make a better decision)

[1]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#logging

Reply:
Same here: I'm sure this was a good decision by then, not so sure the next iteration should be to expose a setting for this instead of using a metric and/or exposing log sampling as part of the collector's telemetry config.

Reply:
@tigrannajaryan any thoughts on the above?

Reply:
I think a possible future option is to supply 2 loggers to every component: one that has no sampling (outputs 100% of logs) and another that is sampled. Components will use the no-sampled logger for everything that is critical to be printed and is known to be non-repeating so it can't flood logs (e.g. startup and shutdown messages. The messages that can repeat for every data item or request will use the sampled logger. The sampled logger can derive its sampling ratio from a central setting (somewhere in the `telemetry` config section).

I do not think merely passing a single sampled logger to every component to be used for every message is good enough since it can result in critical one-off messages being lost.

On a side note: the memory usage by sampled logger that this PR attempts to address is puzzling. Maybe try to figure out why and reduce that instead of disabling sampling?

Reply:
Hi @tigrannajaryan answering the side note. As you can see on the graph https://user-images.githubusercontent.com/5960625/255850931-f9b3e891-2695-4aac-b8d6-9d60cb0dfb9d.png the memory problem is coming from the logger sampling. We also tried modifying the configuration of `NewSamplerWithOptions` 
```
zapcore.NewSamplerWithOptions(
			core,
			10*time.Second,
			1,
			100,
		)
```
But it was still a high memory consumption.
So, it would be great to disable the logger sampler somehow from the `telemetry` config (in a long-term solution) or from the `exporterHelper` config (short-term solution)

Thank you.

Reply:
I agree with @tigrannajaryan. This doesn't seem like a good option to expose in the exporters' configuration interface.

@antonjim-te can you please try initializing one sampled logger instance per service and see if that helps with memory utilization? It can be created in `service.New` function and passed as another field of `TelemetrySettings`  to all components. If it doesn't help with memory utilization, we can still expose a similar configuration for the sampled logger in the `service::telemetry`.

Reply:
Hi @dmitryax ,
Thank you for your suggestion. I will work on that in a separate branch, test it and if it solves the problem I will bring those changes to into this PR.

I will create a ticket in our internal process and add it to our next sprint.

I have a good feeling about this suggestion because it will just create one logger sampled for all the exporters instead of creating one per `exporterHelper`.

Thank you, team. I will keep you updated.

Reply:
Hi team, 
I have followed the suggestion from @dmitryax and created an extra sampled logger in the `TelemetrySettings` and used it when necessary in our case in the `QueuedRetrySender`

Please take a look at the new PR changes.

Thank you in advance.